x-pack/test/api_integration/apis/entity_manager/search.ts

```diff
@@ -0,0 +1,456 @@
+              'agent.name': 'agent-k',
+            },
+            {
+              timestamp: moment(now).subtract(5, 'minute').toISOString(),
+              hostname_field: 'host-four',
+            },
+          ],
+        }),
+      ]);
+
+      await Promise.all([
+        createEntitySourceDefinition(supertest, {
+          source: {
+            id: 'hosts-with-agents-1',
+            type_id: 'hosts-with-agents',
+            index_patterns: ['index-1-with-hosts'],
+            identity_fields: ['host.name'],
+            metadata_fields: [],
+            filters: [],
+            timestamp_field: '@timestamp',
+          },
+        }),
+        createEntitySourceDefinition(supertest, {
+          source: {
+            id: 'hosts-with-agents-2',
+            type_id: 'hosts-with-agents',
+            index_patterns: ['index-2-with-hosts'],
+            identity_fields: ['hostname_field'],
+            metadata_fields: [],
+            filters: [],
+            timestamp_field: 'timestamp',
+          },
+        }),
+      ]);
+
+      const { entities, errors } = await searchEntities(supertest, {
+        type: 'hosts-with-agents',
+        start: moment(now).subtract(10, 'minute').toISOString(),
+        end: moment(now).toISOString(),
+        metadata_fields: ['agent.name'],
+      });
+
+      expect(errors).toEqual([]);
+      expect(entities).toEqual([
+        {
+          'entity.id': 'host-uno',
+          'entity.display_name': 'host-uno',
+          'entity.type': 'hosts-with-agents',
+          'entity.last_seen_timestamp': moment(now).subtract(1, 'minute').toISOString(),
+          'host.name': 'host-uno',
+          hostname_field: 'host-uno',
+          'agent.name': expect.arrayContaining(['agent-a', 'agent-b', 'agent-1']),
+        },
+        {
+          'entity.id': 'host-dos',
+          'entity.display_name': 'host-dos',
+          'entity.type': 'hosts-with-agents',
+          'entity.last_seen_timestamp': moment(now).subtract(2, 'minute').toISOString(),
+          'host.name': 'host-dos',
+          hostname_field: 'host-dos',
+          'agent.name': expect.arrayContaining(['agent-3', 'agent-k']),
+        },
+        {
+          'entity.id': 'host-tres',
+          'entity.display_name': 'host-tres',
+          'entity.type': 'hosts-with-agents',
+          'entity.last_seen_timestamp': moment(now).subtract(4, 'minute').toISOString(),
+          'host.name': 'host-tres',
+          'agent.name': null,
+        },
+        {
+          'entity.id': 'host-four',
+          'entity.display_name': 'host-four',
+          'entity.type': 'hosts-with-agents',
+          'entity.last_seen_timestamp': moment(now).subtract(5, 'minute').toISOString(),
+          hostname_field: 'host-four',
+          'agent.name': null,
+        },
+      ]);
+
+      await Promise.all(deleteIndices.map((fn) => fn()));
+    });
+
+    it('resolves entities with multiple identity fields', async () => {
```

Comment:
Feels like we should test this case across multiple sources as well, WDYT?

Reply:
I'll add